src/getting-started/tooling/esp-generate.md

```diff
@@ -1 +1,101 @@
-# esp-generate
+# `esp-generate`
+
+[`esp-generate`][esp-generate] is a template generation tool that assists users in creating a functional project with most of their desired configurations pre-applied.
+
+## Instalaltion
+To install `esp-generate`:
+
+```shell
+cargo install esp-generate --locked
+```
+
+You can also directly download pre-compiled [release binaries][release-binaries] or use [`cargo-binstall`][cargo-binstall].
+
+[release-binaries]: https://github.com/esp-rs/esp-generate/releases
+[cargo-binstall]: https://github.com/cargo-bins/cargo-binstall
+
+## Generating a Project
+
+To launch the TUI and select your configuration options, run:
+
+```shell
+esp-generate --chip esp32c3 your-project
+```
+
+Replace the chip and project name accordingly.
+
+If you prefer not to use the TUI, you can specify options directly with the `-o/--options` flag. Additionally, using the `--headless` flag will prevent the TUI from launching:
+
+```shell
+esp-generate --chip esp32 -o alloc -o wifi --headless your-project
+```
+
+Adjust the options as needed for your project. See [Available Options][available-options] section of the README.
+
+The tool will generate the following files:
```

Comment:
Just a passing thought, but not sure how much detail we want to go into with regards to the structure and configuration options of the template used in `esp-generate`. Any changes there will require updates to the book, whereas if we just document this in the `esp-generate` repo instead and link to that, the book does not require updating. No real strong feelings here, just might be nice to avoid changing the book in the future as much as possible.